packages/echo/src/handler.ts

```diff
@@ -168,9 +173,39 @@ export class EchoRequestHandler<Input extends any[] = any[], Output = any> {
     workflowId: string,
     stepId: string,
     action: string,
-    anonymousHeader: string,
-    source: string
+    source: string,
+    headers: { originHeader: string; anonymousHeader: string }
   ): Record<PostActionEnum, () => Promise<IActionResponse>> {
+    const { originHeader, anonymousHeader } = headers;
+    const isProduction = process.env.NODE_ENV === 'production';
+    const isValidOrigin =
+      (originHeader && originHeader.includes('https://web.novu.co')) ||
```

Comment:
@denis-kralj-novu 's Framework PR [here](https://github.com/novuhq/novu/pull/5717/files#diff-87afb28697d5d6cf6e6c16619597918fa4ed23adf01747be76412590624eb0b3R91) is using the default API header. I suggest we create a `NovuApiUrlEnum` with both options, then we can simplify this check to:
```typescript
const isValidOrigin = Object.values(NovuApiUrlEnum).includes(originHeader);
```